go-controller/hack/update-codegen.sh

```
@@ -73,8 +77,41 @@ for crd in ${crds}; do
+
+  # Show the API-linter report, but fail if it contains any violation other than
+  # the expected apimachinery (and sometimes our own) names_match ones.
+  if grep -vE '^API rule violation: names_match,' "${new_report}" | grep -q '[^[:space:]]'; then
+    echo "ERROR: openapi-gen reported unexpected API rule violations for ${crd} (only names_match is allowed):" >&2
+    grep -vE '^API rule violation: names_match,' "${new_report}" >&2
+    exit 1
+  fi
+
   echo "Generating apply configuration for $crd ($api_version)"
   applyconfiguration-gen \
+    --openapi-schema <(go run github.com/ovn-kubernetes/ovn-kubernetes/go-controller/pkg/crd/$crd/${api_version}/openapi/cmd/models-schema) \
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)